analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.ReaderWriterLock.cs

```diff
@@ -165,4 +165,18 @@ public void Method10()
         readerWriterLock.AcquireReaderLock(42); // Compliant
         readerWriterLock.ReleaseReaderLock();
     }
+
+    public void WrongOrder()
+    {
+        readerWriterLock.ReleaseReaderLock();
```

Comment:
_Optional:_ After merge of #5340 and rebase, you can use more authentic examples with the `if(condition)`